tests/platform_tests/test_advanced_reboot.py

```diff
@@ -29,9 +30,29 @@
 logger = logging.getLogger()
 
 
+def check_if_ssd(duthost):
+    try:
+        output = duthost.command("lsblk -d -o NAME,ROTA")
+        lines = output['stdout'].strip().split('\n')
+        for line in lines[1:]:
+            name, rota = line.split()
+            if name.startswith('sd') and int(rota) == 0:
+                return True
+        return False
+    except Exception as e:
+        logger.error(f"Error while checking SSD: {e}")
+        return False
+
+
 @pytest.fixture(scope="module", params=[SINGLE_TOR_MODE, DUAL_TOR_MODE])
-def testing_config(request, tbinfo):
+def testing_config(request, duthosts, rand_one_dut_hostname, tbinfo):
     testing_mode = request.param
+    duthost = duthosts[rand_one_dut_hostname]
+    tbname, _ = get_tbinfo(request)
+    is_ssd = check_if_ssd(duthost)
+    neighbor_type = request.config.getoption("--neighbor_type")
+    if '7050' in tbname and not is_ssd and neighbor_type == 'eos':
```

Comment:
Change this to "7050cx3". Also use the platform check. TB name does not necessarily be a unique identifier.
`if "x86_64-arista_7050cx3_32s" in duthost.facts['platform']:`